src/chains.rs

```diff
@@ -92,12 +92,13 @@ fn format_chain_item(
     context: &RewriteContext<'_>,
     rewrite_shape: Shape,
     allow_overflow: bool,
-) -> Option<String> {
+) -> RewriteResult {
+    // TODO this should only match RewriteError::ExceedsMaxWidth
```

Comment:
I think it would be worth giving some additional context here. Are you saying that the `.unknown_error()` call below should be a `.max_width_error()` call instead? If that's the case, it would be useful to know what limitations are preventing us from calling `.max_width_error()`.

Reply:
My original intention was a bit different from what you mentioned, but I think it's worth discussing the use of `.unknown_error()` here. The only reason for returning `None` in this case is when the given span is out of the range covered by the snippet provider in self.context. To express this explicitly, I considered adding another variant like `SpanOutRange`, but I'm not sure how often we would use this variant. That's why I left it as `unknown_error` for now.

Reply:
Thanks for the explanation. If you could capture these details and include them in the TODO comment that would be great so that you or someone else knows how to move forward on this one later on.